lib/logstash/outputs/tcp.rb

```diff
@@ -204,7 +176,38 @@ def server?
   end # def server?
 
   public
-  def receive(event)
-    @codec.encode(event)
-  end # def receive
+  def multi_receive_encoded(encoded)
```

Comment:
I've been naming this variable `events_and_encoded`. I'm not saying it should be named the exact same thing here, but `encoded` to me sounds like a bytes/strings only.